src/mlpack/methods/ann/layer/batch_norm_impl.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file batch_norm_impl.hpp
+ * @author Praveen Ch
+ * @author Manthan-R-Sheth
+ *
+ * Implementation of the Batch Normalization Layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_BATCHNORM_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_BATCHNORM_IMPL_HPP
+
+// In case it is not included.
+#include "batch_norm.hpp"
+
+namespace mlpack {
+namespace ann { /** Artificial Neural Network. */
+
+template<typename InputDataType, typename OutputDataType>
+BatchNorm<InputDataType, OutputDataType>::BatchNorm():
+  size(10),
+  eps(1e-7),
+  deterministic(false)
```

Comment:
Sorry for all the really picky comments here... maybe we need to update the style checker's rules to make sure it catches everything... :)

In this bit there should be a space between `BatchNorm()` and `:`: `BatchNorm() :`, and the initialization list should be tabbed twice.  This happens a few other cases but I don't want to leave 10,000 comments so I'll just make a note of it once here.